packages/block-editor/src/components/block-navigation/dropdown.js

```diff
@@ -80,9 +80,8 @@ function BlockNavigationDropdown(
 					isEnabled={ isEnabled }
 				/>
 			) }
-			renderContent={ ( { onClose } ) => (
+			renderContent={ () => (
 				<BlockNavigation
-					onSelect={ onClose }
```

Comment:
This is actually a bug in trunk that is highlighted by this PR. (In a failing e2e test, focus was moving back to "inner block" after the parent was selected)